File: internal/score/mirrors.go
package score

import (
	"net/url"
	"strings"
)

// feedMirrors lists alternate feed URLs when the publisher host blocks
// datacenter crawl egress (same content, reachable host).
var feedMirrors = map[string][]string{
	"css-tricks.com": {"https://feeds.feedburner.com/CssTricks"},
}

// FeedMirrorsFor returns known mirrors for a feed or page URL's host.
func FeedMirrorsFor(xmlURL string) []string {
	u, err := url.Parse(xmlURL)
	if err != nil {
		return nil
	}
	host := strings.TrimPrefix(strings.ToLower(u.Hostname()), "www.")
	out := feedMirrors[host]
	if len(out) == 0 {
		return nil
	}
	cp := make([]string, len(out))
	copy(cp, out)
	return cp
}

File: web/src/score/mirrors.ts
/** Known publisher mirrors — keep in sync with workers/score/src/mirrors.ts */

const FEED_MIRRORS: Record<string, readonly string[]> = {
  'css-tricks.com': ['https://feeds.feedburner.com/CssTricks'],
}

/** Alternate feed URLs when the publisher host blocks Score egress. */
export function feedMirrorsFor(xmlUrl: string): string[] {
  try {
    const host = new URL(xmlUrl).hostname.replace(/^www\./i, '').toLowerCase()
    return [...(FEED_MIRRORS[host] ?? [])]
  } catch {
    return []
  }
}
